anchore_engine/analyzers/manager.py

```diff
@@ -29,8 +29,8 @@ def apply_hints(analyzer_report, unpackdir):
     # the config does not explicitly enable hints processing, effectively disabling
     # hints processing.
     for engine_entry in utils.content_hints(unpackdir=unpackdir):
-        pkg_type = engine_entry.get("type")
-        if pkg_type:
+        pkg_type = engine_entry.get("type", "")
+        if pkg_type and pkg_type != "go":
```

Comment:
I see why this was throwing an error for `go` before. But a better check would be to test if pkg_type key exists in syft.modules_by_engine_type. Future syft integrations will add more package types. Limiting on a certain package type (such as go) will mean it has to be changed once a mapper for that package type is available

```suggestion
        if pkg_type and pkg_type in syft.modules_by_engine_type:
```

Reply:
fixed